tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -14088,11 +14096,10 @@ add_debug_cmnblk_variables(LL_DebugInfo *db, SPTR sptr)
         debug_name = new_debug_name(SYMNAME(scope), SYMNAME(sptr),
                                     SYMNAME(var));
       }
-      if (gbl.rutype != RU_BDATA && 
-          NEEDMODG(scope) && lookup_modvar_alias(var)) {
+      if (gbl.rutype != RU_BDATA && NEEDMODG(scope) && has_alias &&
+          (!RESTRICTEDG(sptr) || lookup_modvar_alias(var))) {
```

Comment:
I think the intention here would be clearer if the conditions were reorganized like this (which should be logically equivalent):
```
      if (gbl.rutype != RU_BDATA && NEEDMODG(scope) &&
          (lookup_modvar_alias(var) ||
           (has_alias && !RESTRICTEDG(sptr)))) {
```

Reply:
Yes, this looks far better. I am updating it.